src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -295,6 +307,68 @@ public class HubEndPoint<THub, TClient> : EndPoint, IInvocationBinder where THub
             }
         }
 
+        private bool IsChannel(Type type, out Type payloadType)
+        {
+            var channelType = type.AllBaseTypes().FirstOrDefault(t => t.IsGenericType && t.GetGenericTypeDefinition() == typeof(ReadableChannel<>));
```

Comment:
When we get C# 8? 😄 We'll need to make this better in the mean time, file an issue.